src/python_testing/TC_IDM_2_2.py

```diff
@@ -0,0 +1,617 @@
+        read_request_6 = await self.default_controller.ReadAttribute(self.dut_node_id, [attribute_list])
+        returned_endpoints = read_request_6[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request_6[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request_6[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 7
+        # TH sends the Read Request Message to the DUT to read all attributes from a cluster at all Endpoints
+        # AttributePath = [[Cluster = Specific ClusterID]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(7, "Send Request Message to read all attributes from one cluster at all endpoints")
+        # endpoint = [Clusters.Objects.Descriptor]
+        read_request_7 = await self.default_controller.ReadAttribute(self.dut_node_id, [descriptor_obj])
+        returned_endpoints = read_request_7[0].keys()
+        for i in range(3):
+            returned_endpoints = read_request_7[i].keys()
+            # Check if chip.clusters.Objects.Descriptor is in output
+            asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+            # Check if AttributeList is in nested output
+            asserts.assert_in(attribute_list, read_request_7[i][descriptor_obj], "AttributeList not in output")
+
+        # Step 8
+        # TH sends the Read Request Message to the DUT to read all attributes from all clusters at one Endpoint
+        # AttributePath = [[Endpoint = Specific Endpoint]]
+        # On receipt of this message, DUT should send a report data action with the attribute value from all the Endpoints to the DUT.
+        self.print_step(8, "Send Request Message to read all attributes from all clusters at one endpoint")
+        read_request_8 = await self.default_controller.ReadAttribute(self.dut_node_id, [0])
+        returned_endpoints = read_request_8[0].keys()
+
+        # Check if chip.clusters.Objects.Descriptor is in output
+        asserts.assert_in(descriptor_obj, returned_endpoints, "Descriptor cluster not in output")
+        # Check if ServerList is in nested output
+        asserts.assert_in(server_list_attr, read_request_1[0][descriptor_obj], "ServerList not in output")
+
+        # Step 9
+        # TH sends the Read Request Message to the DUT to read an attribute of data type bool.
+        # If the device does not have an attribute of data type bool, skip this step.
+
+        # Verify on the TH that the DUT returns data from the expected attribute path.
+
+        self.print_step(9, "Read a read request of attribute type bool")
+        await self.get_cluster_from_type(bool)
```

Comment:
this is a weird function name because it's not a getter since it doesn't return anything. 

maybe check_attribute_read_for_type ?

Reply:
Fixed locally and will appear in future commit